Framework/WorkflowAlgorithms/src/EQSANSLoad.cpp

```diff
@@ -343,12 +332,13 @@ void EQSANSLoad::moveToBeamCenter() {
       dataWS->getInstrument()->getNumberParameter("number-of-x-pixels")[0]);
   int ny_pixels = static_cast<int>(
       dataWS->getInstrument()->getNumberParameter("number-of-y-pixels")[0]);
-  V3D pixel_first = dataWS->getInstrument()->getDetector(0)->getPos();
+  const auto &detInfo = dataWS->detectorInfo();
+  const V3D pixel_first = detInfo.position(0);
```

Comment:
The methods of `DetectorInfo` take a detector index, not a detector ID (similar to the differecne between workspace index and spectrum number). Unless you can show that ID 0 is always index 0 (this would be the case if 0 is the smallest detector ID in the instrument), you need to convert the ID to an index. You can use `DetectorInfo::indexOf(detid_t)`.